release-notes/swan-lake-beta3-release-note.md

```diff
@@ -34,11 +34,316 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
+ 
+   if x is error {
+       io:println(x);
+   }
+}
+```
+
+##### Wildcard Binding Pattern Support in Variable Declarations
+
+The wildcard binding pattern can now be used in a variable declaration with a value that belongs to type `any`.
+
+```ballerina
+import ballerina/io;
+ 
+float _ = 3.14;
+var _ = io:println("hello");
+```
+
+Using the wildcard binding pattern when the value is an error will result in a compilation error.
+
+```ballerina
+var _ = error("custom error"); // Compilation error.
+```
+
+##### Relaxed Static Type Requirements for `==` and `!=`
+
+Previously, `==` and `!=` were allowed only when both operands were of static types that are subtypes of `anydata`. This has now been relaxed to allow `==` and `!=` if the static type of at least one operand is a subtype of `anydata`.
+```ballerina
+error? x = ();
+ 
+// Now allowed.
+if x == () {
+ 
+}
+``` 
+
 #### Bug Fixes
 
-To view bug fixes, see the [GitHub milestone for Swan Lake Beta3](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+Beta3%22+label%3AType%2FBug+label%3ATeam%2FCompilerFE).
+- In a stream type `stream<T, C>`; the completion type `C` should always include nil if it is a bounded stream. A bug where this was not validated for stream implementors has been fixed.
+
+```ballerina
+class StreamImplementor {
+   public isolated function next() returns record {|int value;|}|error? {
+       return;
+   }
+}
+ 
+stream<int, error> stm = new (new StreamImplementor()); // Will now result in an error.
+```
+
+- Resource methods are no longer added to the type via object type inclusions. This was previously added even though resource methods do not affect typing.
+
+```ballerina
+service class Foo {
+   resource function get f1() returns string {
+       return "foo";
+   }
+ 
+   function f2() returns int => 42;
+}
+ 
+// It is no longer required to implement the `get f1` resource method.
+service class Bar {
+   *Foo;
+ 
+   function f2() returns int => 36;
+}
+```
+
+- A bug in string unescaping of unicode codepoints > `0xFFFF` has been fixed.
+
+```ballerina
+import ballerina/io;
+ 
+public function main() {
+  string str = "Hello world! \u{1F600}";
+  io:println(str);
+}
+```
+
+The above code snippet which previously printed `Hello world!  ὠ0` will now print `Hello world! 😀`.
+
+- A bug in escaping `NumericEscape` has been fixed.
+
+```ballerina
+import ballerina/io;
+ 
+public function main() {
+  string str = "\\u{61}pple";
+  io:println(str);
+}
+```
+
+This code snippet, which previously printed `\u0061pple` will now print `\u{61}pple`.
+
+- A bug that resulted in `NumericEscape` in the template string not being interpreted literally has been fixed.
+
+```ballerina
+import ballerina/io;
+ 
+public function main() {
+  string str = string `\u{61}pple`;
+  io:println(str);
+}
+```
+
+The code snippet above, which previously printed `\u0061pple` will now print `\u{61}pple`.
+
```

Comment:
- A bug that resulted in self-referencing not being detected when referenced via a `let expression` or a `constant reference expression` has been fixed. 
```ballerina
const int CONST1 = CONST1; // Compilation error
public function main() {
    string s = let string[] ar = [s] in ar[0]; // Compilation error
}
```

Reply:
@rdhananjaya can you please check this?

Reply:
+1

Reply:
How about

````suggestion

- A bug that resulted in self-referencing not being detected when referenced via a `let` expression or a constant reference expression has been fixed.

The following will now result in errors.

```ballerina
const int INTEGER = INTEGER; // Compilation error.

public function main() {
    string s = let string[] arr = [s] in arr[0]; // Compilation error.
}
```
````

Reply:
+1